global/scripts/RSNregression.m

```diff
@@ -178,6 +178,56 @@ function RSNregression(InputFile, InputVNFile, Method, ParamsFile, OutputBeta, v
             
             betaICAone = weightedDualRegression(GroupMapcii.cdata, inputConcat, AreaWeights .* ScaledAlignmentQuality);
             [betaICA, NODEts] = weightedDualRegression(normalise(betaICAone), inputConcat, AreaWeights);
+            
+            if strcmp(Method, "tICA_weighted") % based on tICA/scripts/ComputeGroupTICA.m but for one subject as a group
+                NODEts=NODEts';
+                disp(['NODEts'])
+                size(NODEts)
+                SpectraArray = textscan(optional.SpectraParams, '%s', 'Delimiter', {'@'});
+                nTPsForSpectra =str2double(SpectraArray{1}{1});
+                thisStart = 1;
+                TCSRunVarSub = [];
+                for i = 1:size(inputArray, 1)
+                    dtseriesName=[inputArray{i}];
+                    if exist(dtseriesName, 'file')
+                        [~, runLengthStr] = system(['wb_command -file-information -only-number-of-maps ' dtseriesName]);
```

Comment:
Consider updating `my_system` below to return the stdout, and using it, which avoids passing matlab's library env variables to other executables.